### PR TITLE
add utility function for grabbing resources by tag

### PR DESCRIPTION
**Pull request by tedivm**

### CODE REVIEW

File: test/utils/tags.go [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/kms"
+	"github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi"
+	tagging_types "github.com/aws/aws-sdk-go-v2/service/resourcegroupstaggingapi/types"
+	"github.com/aws/aws-sdk-go-v2/service/secretsmanager"
+)
+
+const maxPages = 10
+
+// GetTaggedResources -
```
Comment (madhuravius) [inline test/utils/tags.go:17]: might wanna populate comment or drop it as to what this does

File: test/utils/tags.go [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+	params := &resourcegroupstaggingapi.GetResourcesInput{
+		TagFilters: []tagging_types.TagFilter{
+			{Key: aws.String("aptible_asset_id"), Values: []string{assetId}},
+			{Key: aws.String("aptible_environment_id"), Values: []string{environmentId}},
+		},
+	}
+
+	for i := 0; i < maxPages; i++ {
+		response, err := c.GetResources(ctx, params)
```
Comment (madhuravius) [inline test/utils/tags.go:37]: i'd probably either do a logger statement if max pages is hit OR throw an error that pagination limit exceeded, not sure which so people know the results list is possibly truncated (though unlikely)

**Comment by madhuravius** [comment]
prob wanna rename `resource_arn` => `resourceARN` or `resourceArn` or something

File: test/basic/vpc_create/vpc_create_test.go [hunk 3]
```diff
@@ -13,8 +14,18 @@ import (
 
 	cac "github.com/aptible/cloud-api-clients/clients/go"
 	"github.com/aptible/terraform-provider-aptible-iaas/internal/client"
+	"github.com/aptible/terraform-provider-aptible-iaas/test/utils"
 )
 
+func cleanupAndAssert(c context.Context, t *testing.T, terraformOptions *terraform.Options, environmentId, assetId string) {
```
Comment (madhuravius) [inline test/basic/vpc_create/vpc_create_test.go:20]: nitpick, might wanna keep using the same name `ctx` everywhere, i think we use the var `c` for client in some places

File: test/basic/vpc_create/vpc_create_test.go [hunk 4]
```diff
@@ -13,8 +14,18 @@ import (
...
+	terraform.Destroy(t, terraformOptions)
+
+	resources, err := utils.GetTaggedResources(c, environmentId, assetId)
```
Comment (madhuravius) [inline test/basic/vpc_create/vpc_create_test.go:23]: i think you want to use this in a truthy case too, to make sure that it actually captures resources in a length that you expect in the TestVPCCreate, so you'd have something like:

```
assert.Equal(t, 10, len(resources)) // or maybe just a value greater than 0?
```

in that stanza, you'd never know if this doesn't work if you didnt do this

File: test/utils/tags.go [hunk 5]
```diff
@@ -116,8 +118,14 @@ func verifyResourcesIsActive(ctx context.Context, resourceARN string) (bool, err
 		c := ec2.NewFromConfig(cfg)
 
 		// Extract the instance ID since the APIs do not accept the ARN.
-		r := regexp.MustCompile(`instance/(i-[a-f0-9]{17})`)
-		results := r.FindStringSubmatch(resourceARN)
+		// r := regexp.MustCompile(`instance/(i-[a-f0-9]{17})`)
```
Comment (madhuravius) [inline test/utils/tags.go:121]: can prob be deleted

File: test/basic/vpc_create/vpc_create_test.go [hunk 6]
```diff
@@ -101,4 +101,8 @@ func TestVPCCreate(t *testing.T) {
 	assert.Nil(t, networkAnalysisErr)
 	assert.Equal(t, len(networkAnalysis.NetworkInsightsAnalyses), 1)
 	assert.Equal(t, *networkAnalysis.NetworkInsightsAnalyses[0].Status, "succeeded")
+
+	tagged_resources, err := utils.GetTaggedResources(ctx, os.Getenv("ENVIRONMENT_ID"), vpcAsset.Id)
```
Comment (madhuravius) [inline test/basic/vpc_create/vpc_create_test.go:105]: casing consistency nitpick, we should def update our lint rules, something's missing on that, but i'm okay with this and we can revisit these in lump sum to be honest

**Review by madhuravius — APPROVED** [review]
LGTM minor cleanup outstanding at some point, but not critical for the work itself

File: test/basic/vpc_create/vpc_create_test.go (hunk 6 above)
Comment (tedivm) [reply test/basic/vpc_create/vpc_create_test.go:105]: Thanks! My brain is so used to typing python these days that I keep doing this.

**Review by madhuravius — APPROVED** [review]

**Pull request merged**